dietpi/dietpi-software

```diff
@@ -11942,37 +11942,36 @@ _EOF_
 			G_EXEC_OUTPUT=1 G_EXEC docker run -d -p '9002:9000' --name=portainer --restart=always -v '/run/docker.sock:/var/run/docker.sock' -v 'portainer_data:/data' 'portainer/portainer-ce'
 		fi
 
-		if To_Install 141 adsb-setup adsb-docker # ADSB Feeder
+		if To_Install 141 adsb-setup adsb-docker # ADS-B Feeder
 		then
-			# create the userdata directory and clone the adsb-feeder repo into /tmp
-			G_EXEC mkdir -p /mnt/dietpi_userdata/adsb-feeder
+			# clone the adsb-feeder repo into /tmp
 			G_EXEC_OUTPUT=1 G_EXEC git clone -b dietpi 'https://github.com/dirkhh/adsb-feeder-image.git' /tmp/adsb-feeder
 
 			# remove the service that isn't needed for an app install on DietPi and install the rest
 			G_EXEC cd /tmp/adsb-feeder/src/modules/adsb-feeder/filesystem/root
-			G_EXEC rm -rf ./usr/lib/systemd/system/adsb-bootstrap.service
-			G_EXEC cp -a ./usr/lib/systemd/system/* /etc/systemd/system/
+			G_EXEC rm ./usr/lib/systemd/system/adsb-bootstrap.service
+			G_EXEC mv ./usr/lib/systemd/system/* /etc/systemd/system/
 
 			# determine the version
-			ADSB_FEEDER_DATE_COMPONENT=$(git log -20 --date=format:%y%m%d --format="%ad" | uniq -c | head -1 | awk '{ print $2"."$1 }')
-			ADSB_FEEDER_TAG_COMPONENT=$(git describe --match "v[0-9]*" | cut -d- -f1)
-			ADSB_FEEDER_VERSION="${ADSB_FEEDER_TAG_COMPONENT}(dietpi)-${ADSB_FEEDER_DATE_COMPONENT}"
+			local ADSB_FEEDER_DATE_COMPONENT=$(git log -20 --date='format:%y%m%d' --format='%ad' | uniq -c | mawk '{print $2"."$1;exit}')
+			local ADSB_FEEDER_TAG_COMPONENT=$(git describe --match 'v[0-9]*' | cut -d- -f1)
+			local ADSB_FEEDER_VERSION="$ADSB_FEEDER_TAG_COMPONENT(dietpi)-$ADSB_FEEDER_DATE_COMPONENT"
 
 			# create the target directory for the app and populated with the code from the git checkout
 			[[ -d '/opt/adsb' ]] && G_EXEC rm -R /opt/adsb
-			G_EXEC cp -a /tmp/adsb-feeder/src/modules/adsb-feeder/filesystem/root/opt/adsb /opt
+			G_EXEC mv /tmp/adsb-feeder/src/modules/adsb-feeder/filesystem/root/opt/adsb /opt/
 			G_EXEC cd /opt/adsb
 
 			# remove the git clone of the repo we installed from
-			G_EXEC rm -rf /tmp/adsb-feeder
+			G_EXEC rm -R /tmp/adsb-feeder
 
 			# create a symlink so the config files reside where they should be in /mnt/dietpi_userdata/adsb-feeder
 			G_EXEC mkdir -p /mnt/dietpi_userdata/adsb-feeder/config
 			G_EXEC ln -s /mnt/dietpi_userdata/adsb-feeder/config .
 
 			# set the 'image name' and version that are shown in the footer of the Web UI
-			echo "ADSB Feeder app running on DietPi" > feeder-image.name
-			echo "$ADSB_FEEDER_VERSION" > adsb.im.version
+			G_EXEC eval 'echo '\''ADSB Feeder app running on DietPi'\'' > feeder-image.name'
+			G_EXEC eval "echo '$ADSB_FEEDER_VERSION' > adsb.im.version"
```

Comment:
does this work? I thought variables don't get expanded inside single quotes?
I may be out of my shell programming depth here. Not sure why you'd switch this to the eval construct in the first place

Reply:
As the single quotes are wrapped within double quotes, it does work 🙂. `eval` gets the string with the variable expanded already, so what is between the single quotes then can/must be taken literal.